backend/app/websocket/rate_limiter.py

```
@@ -0,0 +1,78 @@
+    def __init__(self, capacity: int, refill_rate: float) -> None:
+        if capacity <= 0:
+            raise ValueError("Capacity must be greater than 0")
+        if refill_rate <= 0:
+            raise ValueError("Refill rate must be greater than 0")
+
+        self.capacity = capacity
+        self.refill_rate = refill_rate
+        self.tokens = float(capacity)
+        self.last_refill = time.time()
+    
+import threading
```

Ruff (0.12.2)

Redefinition of unused `threading` from line 18

Remove definition: `threading`

(F811)